plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/AbstractIcebergTableOperations.java

```diff
@@ -257,4 +264,25 @@ protected static List<Column> toHiveColumns(List<NestedField> columns)
                         Optional.empty()))
                 .collect(toImmutableList());
     }
+
+    protected CommitStatus checkCommitStatus(String metadataLocation)
+    {
+        AtomicReference<CommitStatus> status = new AtomicReference<>(CommitStatus.UNKNOWN);
+
+        Tasks.foreach(metadataLocation)
+                .retry(3)
+                .suppressFailureWhenFinished()
+                .exponentialBackoff(100, 5000, 600000, 4.0)
+                .run(
+                        location -> {
+                            TableMetadata metadata = refresh();
+                            if (metadata != null) {
+                                if (metadata.metadataFileLocation().equals(metadataLocation)
+                                        || metadata.previousFiles().stream().anyMatch(log -> log.file().equals(metadataLocation))) {
```

Comment:
can `metadata.previousFiles()` or `log.file()` be nullable?
(iceberg library is null-friendly)

Reply:
These values will not be nullable.

https://github.com/apache/iceberg/blob/e2bb9ad7e792efca419fa7c4a1afde7c4c44fa01/core/src/main/java/org/apache/iceberg/TableMetadata.java#L1286-L1291

https://github.com/apache/iceberg/blob/1.0.x/core/src/main/java/org/apache/iceberg/TableMetadata.java#L1504-L1506